script.js

```diff
@@ -0,0 +1,442 @@
+    return 1;
+
+  if (operator === '*' || operator === '/') 
+    return 2;
+
+  if (operator === 'sin' || operator === 'cos' || operator === 'tan' || operator === 'sqrt') 
+    return 3;
+
+  if (operator === '^') 
+    return 4;
+  
+  return 0;
+}
+
+function isOperator(token) {
+  return ['+', '-', '*', '/', 'sin', 'cos', 'tan', 'sqrt','^'].includes(token);
+}
+
+function evaluateOperator(operator, operand1, operand2) {
+  switch (operator) {
+    case '+':
+      return (Number(operand1) + Number(operand2)).toFixed(4);
+
+    case '-':
+      return (Number(operand1) - Number(operand2)).toFixed(4);
+
+    case '*':
+      return (Number(operand1) * Number(operand2)).toFixed(4);
+
+    case '/':
+      if(operand2 == 0)
+      {
+        return "Math Error";
+      }
+      return (Number(operand1) / Number(operand2)).toFixed(4);
+
+    case '^':
+        return Math.pow(Number(operand1), Number(operand2));
+
+    case 'sin':
+      return Math.sin(Number(operand2) * (Math.PI / 180)).toFixed(4);
+
+    case 'cos':
+      return Math.cos(Number(operand2) * (Math.PI / 180)).toFixed(4);
+
+    case 'tan':
+      if(operand2 == 90)
+      {
+        return "Undefined";
+      }
+      return Math.tan(Number(operand2) * (Math.PI / 180)).toFixed(4);
+
+    case 'sqrt':
+      return Math.sqrt(Number(operand2)).toFixed(4);
+
+    default:
+      throw new Error('Unsupported operator: ' + operator);
+  }
+}
+
+function infixToPostfix() {
+  const textInput = document.getElementById('textInput');
+  textInput.style.height = "50%";
+  textInput.style.fontSize = "17px";
+  const textInput1 = document.getElementById('answer');
+  textInput1.style.display = "flex";
+  const expression = textInput.value; 
+  const output = [];
+  const operatorStack = [];
+
+  let tokens1;
+
+  w2 =  Array.from(expression);
+  w3 =  Array.from(expression);
+
+  for(var i = 0; i < w3.length ; i++)
+  {
+  var j = 0;
+  
+    if(w3[i] === "(" && w3[j+i+1] === "+")
+    {
+      w3.splice(i,1);
+      const result = w3.join('');
+      w2 = w3;
+
+      if(w3[j+i+2] === ")")
+      {
+        w3.splice(j+i+2,1);
+        const result = w3.join('');
+        w2 = w3;
+      }
+    }
+  }
+
+  if(w2[0] === "+")
+  {
+    w2.splice(0,1);
+    const result = w2.join('');
+    tokens1 = result;
+  }
+  else
+  {
+    tokens1 = w2.join("");
+  }
+
+  tokens = tokens1.match(/(?:sin|cos|tan|sqrt|\b[A-Za-z0-9.]{1,9}(?![\d.])\b|\^|\b[+\-*/()]\b|e|pi|((?:\d+(?:\.\d)?|\.\d+|[+\-*/()])))?/g);
+  const w1 = Array.from(tokens);
+  const w = w1.filter(Boolean);
+
+  if(w[w.length-1] == "+" || w[w.length-1] == "-" || w[w.length-1] == "/" || w[w.length-1] == "*" || w[w.length-1] == "^")
+  {
+    const ans = document.getElementById('answer');
+    ans.value = "Invalid Input";
+    return;
+  }
+
+  if(w[w.length-1] == "(")
+  {
+    const ans = document.getElementById('answer');
+    ans.value = "Invalid Input";
+    return;
+  }
+
+for(var i = 0; i < w.length ; i++)
```

Comment:
- calculate w.length in a variable before and then use it here instead of calculating on and on
- resolve this to use map or forEach instead of for loop